Lib/fontTools/subset/__init__.py

```diff
@@ -2091,17 +2097,14 @@ def subset_glyphs(self, s):
         ]
         table.AttachList.GlyphCount = len(table.AttachList.AttachPoint)
     if hasattr(table, "MarkGlyphSetsDef") and table.MarkGlyphSetsDef:
-        for coverage in table.MarkGlyphSetsDef.Coverage:
+        markGlyphSets = table.MarkGlyphSetsDef
+        for coverage in markGlyphSets.Coverage:
             if coverage:
                 coverage.subset(glyphs)
 
-        # TODO: The following is disabled. If enabling, we need to go fixup all
-        # lookups that use MarkFilteringSet and map their set.
-        # indices = table.MarkGlyphSetsDef.Coverage = \
-        #   [c for c in table.MarkGlyphSetsDef.Coverage if c.glyphs]
-        # TODO: The following is disabled, as ots doesn't like it. Phew...
-        # https://github.com/khaledhosny/ots/issues/172
-        # table.MarkGlyphSetsDef.Coverage = [c if c.glyphs else None for c in table.MarkGlyphSetsDef.Coverage]
+        s.used_mark_sets = [i for i, c in enumerate(markGlyphSets.Coverage) if c.glyphs]
```

Comment:
This introduces a dependency on the order the tables are subset. We don't enforce that right now.

Reply:
I thought `_sort_tables` would sort the tables alphabatically, so `GDEF` always comes first, but reading again it seem to leave the other tables in their original order. May be I should add `GDEF`, `GSUB` and `GPOS` their?
https://github.com/fonttools/fonttools/blob/a537f97b44118e7a14cc6e507701e9fca95b81e5/Lib/fontTools/subset/__init__.py#L3498C4-L3502

Reply:
Sounds right.

Reply:
https://github.com/fonttools/fonttools/pull/3118